### PR TITLE
fix(chrono): improved logic for operations

### LINKED ISSUE #5230: Working clock while in cold and dark state
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-06-28T07:23:55+00:00",
  "ref": "master",
  "sha": "ac6847baf2d64bc836c9793bc224ae0e34910ae6",
  "actor": "davidwalschots",
  "event_name": "manual"
}
```

### Describe the bug

I load the aircraft cold and dark, everything is off as it should, expect the clock is working...

### Expected behavior

I would expect the clock not to be working in a cold and dark state.

### Steps to reproduce

1. Load the A320nx
2. Start a flight at a gate/parking so it is spawned cold and dark
3. Verify the clock is working

### References (optional)

CD

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by sidnov**
Fixes #5230 

## Summary of Changes

Improved overall logic for operation and display of the chronometer.

- Chrono computer works even when the aircraft is completely powered down (being connected to DC HOT BUS 1)
- Chrono display should work only when DC ESS is powered
- The buttons to control chrono and elapsed time work only when DC ESS is powered and become inactive once DC ESS is unpowered
- Chrono and Elapsed time counters once started, keep running, even if the display is off (i.e. connected to DC HOT BUS 1).

## References

Komp

## Additional context

Thanks to @komp1821 for testing out the weird test cases on the real bus, and @davidwalschots for his constant guidance :)

Discord username (if different from GitHub): sidnov#8337

## Testing instructions

1. Spawn cold and dark, observe the chrono display is OFF
2. Turn on both batteries or external power (whichever available). Chrono display should turn on.
3. Test all the basic functions. Chrono, Elapsed Time, Time/Date.
4. Turn on the chronometer and elapsed time, then turn off the batteries/external power to put the aircraft in cold and dark state again. Observe chrono display turning off.
5. Try stopping the chronometer, elapsed time in the powered down state. Turn the batteries/external power on and observe chrono/ET is still running.
6. Refer to the summary of changes to find any edge cases.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/instruments/src/Clock/Components/ElapsedTime.tsx [hunk 1]
```diff
@@ -6,19 +6,22 @@ const getDisplayString = (seconds: number, running: boolean) : string => (second
...
+        if (dcEssIsPowered) {
+            if (elapsedKnobPos === 0) {
+                setElapsedTime(Math.max(elapsedTime + absTime - prevTime, 0));
+            } else if (elapsedKnobPos === 2) {
+                setElapsedTime(0);
+            }
+            setPrevTime(absTime);
         }
-        setPrevTime(absTime);
     }, [absTime, elapsedKnobPos]);
```
Comment (tracernz) [inline src/instruments/src/Clock/Components/ElapsedTime.tsx:25]: This should also depend on dsEssIsPowered, shouldn't it?
```suggestion
    }, [absTime, elapsedKnobPos, dcEssIsPowered]);
```

Comment (sidnov) [reply src/instruments/src/Clock/Components/ElapsedTime.tsx]: Missed that one, and yes, useEffect here would depend on dcEssIsPowered

**Review by davidwalschots — APPROVED** [review]

**Comment by Samueleonard** [comment]
Quality Assurance Tester/Trainee Report

Discord: 1samueleonard1#0001
Object of testing: (#5282)
Tier of Testing: (1)
Date: (08/07/2021)

Testing Process: 
. loaded in C+D - display is off
. powered 1 battery - display still off
. powered both bats - display comes on
. switched bats off + ext pwr on - displays comes on
. played with functions of chrono - all works as before (still no gps/int but not part of this PR?)
. turn on chrono + switch off bats - potential issue below

Negatives: 
.chrono continues to count even in C+D? display is off but when powered up again it continues counting from where it should be (potentially correct behaviour, but not what i would expect, never tried before though)

Testing Results: (Passed)

Conclusions: i would say it passes, as long as the issue above can be clarified, its most likely correct behaviour.
Media: n/a

**Comment by davidwalschots** [comment]
@Samueleonard see the PR description:
> Chrono computer works even when the aircraft is completely powered down (being connected to DC HOT BUS 1)

**Comment by Samueleonard** [comment]
oops, i dont know how i missed that, updated 😄

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #5282
Tier of Testing: 1
Date: 10/07/2021

Testing Process:
- Spawned C&D at EDDP
- Looked for chrono display off
- Turned on bat. > display on
- Bat. off. Turned on ext. power > display on
- Tested chrono functions (also with timelaps)
- Tested chrono functions with power off (also with timelaps)

Negatives: -

Testing Results: Passed
Conclusions: Works
Hope this helps :-)

**Comment by AviationSpotter** [comment]
Quality Assurance Trainee Report

Discord: Aviation Spotter#9102
Object of testing: #5282
Tier of Testing: 1
Date: 10/07/2021

Testing Process:

1. Spawned C&D
2. Followed testing instructions chronologically while observing chrono display

Negatives: N/A
Testing Results: Passed

**Comment by FlightSimmer7700** [comment]
Quality Assurance Tester

Discord: FlightSimmer7700#3468
Object of testing: #5282 
Tier of Testing: 1
Date: 10/7/21

Testing Process: 
Loaded in C&D
Loaded in on runway
Loaded in on approach
Loaded in mid-air
tested chrono in all situations 

Negatives: N/A

Testing Results: Passed

Conclusions: LGTM

**Pull request merged**